internal/driver/internal/k8s/logicalvolume_service.go

```diff
@@ -313,36 +312,39 @@ func (s *LogicalVolumeService) ExpandVolume(ctx context.Context, volumeID string
 		return err
 	}
 
-	// wait until topolvm-node expands the target volume
-	for {
-		logger.Info("waiting for update of 'status.currentSize'", "name", lv.Name)
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		case <-time.After(1 * time.Second):
-		}
-
+	return wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
+		Duration: 1 * time.Second, // initial backoff
+		Factor:   1,               // factor for duration increase
+		Jitter:   0.1,
+		Cap:      300 * time.Second, // wait up to 10 seconds per retry, 1, 2, 4, 8, 10 seconds.
+		Steps:    300,               // run at most 10 times
+	}, func(ctx context.Context) (done bool, err error) {
 		var changedLV topolvmv1.LogicalVolume
-		err := s.getter.Get(ctx, client.ObjectKey{Name: lv.Name}, &changedLV)
-		if err != nil {
+		if err := s.getter.Get(ctx, client.ObjectKey{Name: lv.Name}, &changedLV); err != nil {
 			logger.Error(err, "failed to get LogicalVolume", "name", lv.Name)
-			return err
+			return true, err
```

Comment:
```suggestion
			return false, err
```

Reply:
Just to make sure, previously this did not result in a for loop iteration because the return condition triggered. Do we want to switch to retries here as well?

Reply:
Also the bool variable indicates done, so if we set this to false, then it would keep retrying (different from today)